lib/main.py

```diff
@@ -302,6 +308,14 @@ def fetch_scala(ws, args, agg=True) -> None:
         scalaplugin.fetch_ivy_dependencies(files, install_dir, ivy_cache_dir)
 
 
+# NOTE: we could run commands in parallel but the potential for unexpected behaviour
+# may not be worth it
+def foreach(ws, args) -> None:
+    pkg_folders = [f.parent for f in ws.path.glob('*/wit-manifest.json')]
```

Comment:
I think this is a useful feature, but I wonder about a couple things here --

1. As written this will only operate on packages that contain a `wit-manifest.json` file, but not all packages necessarily have this file (they may not have any dependencies)

2. This may operate on packages that are no longer part of the workspace dependency hierarchy. Maybe that's OK? Maybe not? Not sure...

I feel like it may be safer to get a canonical package list by running the resolution algorithm (or maybe just reading `wit-lock.json`. I think we also should have a `-n` option here that just says which packages *would* be processed, without actually running the command in them. Like:

```
Skipping execution because of -n. Would have run:
cd mypackage1 && git tag foo
cd mypackage2 && git tag foo
cd mypackage3 && git tag foo
```

Reply:
You're right. I think this is pending on #33 (package resolution without changing workspace files), which is pending on a finalized `update` function.